### PR TITLE
tools: Update README.node_modules

### PR DESCRIPTION
**Pull request by martinpitt**
- Clean up unnecessarily complicated grammar/language
- Fix some outright wrong statements like "Cockpit does not use npm packages code at runtime" -- we very clearly do (React/PF).
- `npm-install.yml` is no more; replace with description of reposchutz.
- node-cache.git is shared between all org projects
- Setting up a new one from scratch only applies to forks or other orgs. Otherwise the key deployment is driven by a bots script.

Preview: https://github.com/martinpitt/cockpit/blob/node-docs/tools/README.node_modules.md

### CODE REVIEW

File: tools/README.node_modules.md [hunk 1]
```diff
@@ -1,12 +1,10 @@
...
-Cockpit uses Node.js dependencies to process its code. The required Node.JS
-javascript code is packaged in the `node_modules/` directory when distributed.
+Cockpit uses Node.js and [npm](https://www.npmjs.com/) dependencies to build
+its code and for runtime JavaScript libraries.
 
-Cockpit does not use Node.js or any npm packages code at runtime. In addition
-when building straight from a tarball it is also not necessary to have Node.js
-installed.
+Cockpit does not use npm or Node.js at runtime. When building/installing from
+a release tarball it is also not necessary to have Node.js installed.
```
Comment (allisonkarlitskaya) [inline tools/README.node_modules.md:7]: Maybe you want to press the point that we *do* use the modules at runtime, though?

**Review by allisonkarlitskaya — APPROVED** [review]
Looks good, but maybe a minor tweak?

File: tools/README.node_modules.md (hunk 1 above)
Comment (martinpitt) [reply tools/README.node_modules.md]: Hmm, that's what the "and for runtime JavaScript libraries." in the previous paragraph was about. I changed this whole section again, hopefully more clear now?

**Review by allisonkarlitskaya — APPROVED** [review]
This is substantially more explicit, thanks!

**Pull request merged**